packages/babel-core/src/transformation/util/clone-deep.ts

```diff
@@ -1,33 +1,73 @@
-//https://github.com/babel/babel/pull/14583#discussion_r882828856
-function deepClone(value: any, cache: Map<any, any>): any {
+const circleSet = new Set();
+let depth = 0;
+// https://github.com/babel/babel/pull/14583#discussion_r882828856
+function deepClone(
+  value: any,
+  cache: Map<any, any>,
+  allowCircle: boolean,
+): any {
   if (value !== null) {
-    if (cache.has(value)) return cache.get(value);
+    if (allowCircle) {
+      if (cache.has(value)) return cache.get(value);
+    } else if (++depth > 100) {
```

Comment:
I was trying to see what are reasonable depth values: both `@babel/standalone/babel.js` and `@babel/standalone/babel.min.js` are around 200.

Let's increase this to 200 at least, maybe 250 to be safe? It's unlikely that there are files deeper than that.

(Or maybe we could also just completely remove the cycle-detection logic and just throw an error if `depth > 1000` 😬)

Reply:
Thanks for your investigation!
I'd be inclined to set it to 250, since there are probably a few hundred variables being concatenated(auto-generated code)  :)